File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        n = len(s)

        left = 0 
        seen = set()
        maxLen = 0

        for right in range(n):
            while s[right] in seen:
                seen.remove(s[left])
                left += 1

            seen.add(s[right])
            maxLen = max(maxLen, right-left+1)

        return maxLen
            


            

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int left = 0;
        int maxLength = 0;

        for (int right = 0; right < s.length(); right++){
            while (s.substring(left,right).contains(String.valueOf(s.charAt(right)))){
                left += 1;
            }
            maxLength = Math.max(maxLength, right-left+1);
        }
        return maxLength;
        
    }
}